src/vma/util/utils.cpp

```diff
@@ -211,6 +212,47 @@ unsigned short compute_tcp_checksum(const struct iphdr *p_iphdr, const uint16_t
       return (unsigned short)sum;
 }
 
+/* set udp checksum: given IP header and UDP datagram
+ *
+ * (assume checksum field in UDP header contains zero)
+ * This code borrows from other places and their ideas.
+ */
+
+unsigned short compute_udp_checksum(const struct iphdr *p_iphdr, const uint16_t *p_ip_payload) {
+    register unsigned long sum = 0;
+    struct udphdr *udphdrp = (struct udphdr*)(p_ip_payload);
+    unsigned short udp_len = htons(udphdrp->len);
+
+    //add the pseudo header
+    sum += (p_iphdr->saddr>>16)&0xFFFF;
```

Comment:
Please consider using space to separate operands here and following several lines.